src/git.rs

```diff
@@ -13,28 +13,37 @@ pub struct Git {
 }
 
 impl Git {
-    /// Creates `Git` client from environmesnt variables
-    ///
-    /// # Panics
-    ///
-    /// If `CARGO_TAG_EMAIL` or `CARGO_TAG_NAME` is not set
-    pub fn from_env(branch: &str) -> Self {
-        let email = std::env::var("CARGO_TAG_EMAIL").expect("CARGO_TAG_EMAIL not set");
-        let name = std::env::var("CARGO_TAG_NAME").expect("CARGO_TAG_NAME not set");
-
-        Git::open(branch, &email, &name).expect("Failed to open Git repository")
-    }
-
-    /// Opens the Git repository in the current working directory and uses the
-    /// provided `email`, `name` and `branch` to perform Git operations like
-    /// `commit` and `tag`.
-    pub fn open(branch: &str, email: &str, name: &str) -> Result<Self, Box<dyn std::error::Error>> {
+    /// Opens the Git repository in the current working directory and
+    /// uses the provided `branch` to perform Git operations like
+    /// `commit` and `tag`. The `email` and `name` to be used are
+    /// retrieved from git config unless overridden by
+    /// `CARGO_TAG_EMAIL` and `CARGO_TAG_NAME` environment variables.
+    pub fn open(branch: &str) -> Result<Self, Box<dyn std::error::Error>> {
+        let cfg = Config::open_default()?;
         let cwd = current_dir()?;
         let repo = Repository::open(cwd)?;
 
+        let name = if let Ok(name) = std::env::var("CARGO_TAG_NAME") {
+            name
+        } else {
+            cfg.get_entry("user.name")?
+                .value()
+                .ok_or("user.name not utf8")?
+                .to_string()
+        };
+
+        let email = if let Ok(email) = std::env::var("CARGO_TAG_EMAIL") {
+            email
+        } else {
+            cfg.get_entry("user.email")?
+                .value()
+                .ok_or("user.email not utf8")?
+                .to_string()
+        };
```

Comment:
Looks very good! I would split both approaches, one to initialize from env and another to initialize from Git Config.
This is to have predictable behavior, this allows us to define the expected procedure and avoid making assumptions.

So if the users executes the CLI as:

```bash
cargo tag --env [...]
```

Then we would initialize the client using `from_env`.

Instead if no `--env` is used, we can use the `git config` approach here.

WDYT?

Reply:
So then you would insist on getting both name and email from env or both from config? I was thinking this way it would get both from config, but you could override either one with env.

Reply:
We wont override.
We either get from `git config` or from env. But we wont use both at the same time.

Reply:
- If we run: `cargo tag --env` Then we build Git client using env vars.
- If we run `cargo tag` Then we use `git config`
- If we run `cargo tag` and theres no `user.name` nor `user.email` then we just abort (panic), this is because we dont want to take creds from env unless it is explicitly specified.

Reply:
Hm. That seems more restrictive for no reason.

- As it is now, `cargo tag` uses `git config`.

- `CARGO_TAG_NAME=foo CARGO_TAG_EMAIL=foo@bar cargo tag` uses environment variables.

- `CARGO_TAG_NAME=foo cargo tag` uses both.

- `CARGO_TAG_EMAIL=foo@bar cargo tag` uses both.

There is no need for a separate option to use the environment. But if that's what you want, I suppose we can do that.

Reply:
The goal is to minimize side effects.

Reply:
Sorry, I thought that was the whole point of environment variables. If you want to use a command line option for `--env`, then why not just have options for `--name` and `--email` and skip the environment variables?